### ISSUE TITLE
AkkaHttpServer handler might fail with NPE in handleRequest

### ISSUE THREAD

**Issue by jrudolph** [playframework/playframework#10107]
Originally report by @juli1: https://github.com/akka/akka-http/issues/2979
Play 2.8.0

```
a.a.ActorSystemImpl - Internal server error, sending 500 response
java.lang.NullPointerException: null
at play.core.server.AkkaHttpServer.handleRequest(AkkaHttpServer.scala:299)
at play.core.server.AkkaHttpServer.$anonfun$createServerBinding$1(AkkaHttpServer.scala:186)
at datadog.trace.instrumentation.akkahttp.AkkaHttpServerInstrumentation$DatadogAsyncWrapper.apply(AkkaHttpServerInstrumentation.java:176)
at datadog.trace.instrumentation.akkahttp.AkkaHttpServerInstrumentation$DatadogAsyncWrapper.apply(AkkaHttpServerInstrumentation.java:159)
at akka.stream.impl.fusing.MapAsync$$anon$30.onPush(Ops.scala:1261)
```

The line in question: 

https://github.com/playframework/playframework/blob/3c008185a851c26c8cf82ee46ddb0882086fce7b/transport/server/play-akka-http-server/src/main/scala/play/core/server/AkkaHttpServer.scala#L299

Probably an initialization and/or visibility problem. Since `createServerBinding` is run in the constructor from a line above the line where `requestIds` is initialized the binding might "overtake" the remaining constructor.

Overall it is dangerous to run side-effects like `createServerBinding` from a field initializer if other fields might be accessed concurrently right after or while the constructor is still running.

**Comment by kajmagnus** [comment]
This error happens to me when a HTTP request arrives *whilst Play is starting*. Otherwise I think I've never seen it.

`addSbtPlugin("com.typesafe.play" % "sbt-plugin" % "2.8.1")`

Here's my full call stack:

```
{"eventTime":"2020-05-28T08:42:03.860Z",
"message":"Internal server error, sending 500 response\nj
ava.lang.NullPointerException\n\
 at  play.core.server.AkkaHttpServer.handleRequest(AkkaHttpServer.scala:303)\n\
 at  play.core.server.AkkaHttpServer.$anonfun$createServerBinding$1(AkkaHttpServer.scala:190)\n\
 at  akka.stream.impl.fusing.MapAsync$$anon$30.onPush(Ops.scala:1289)\n\
 at  akka.stream.impl.fusing.GraphInterpreter.processPush(GraphInterpreter.scala:541)\n\
 at  akka.stream.impl.fusing.GraphInterpreter.execute(GraphInterpreter.scala:423)\n\
 at  akka.stream.impl.fusing.GraphInterpreterShell.runBatch(ActorGraphInterpreter.scala:624)\n\
 at  akka.stream.impl.fusing.GraphInterpreterShell$AsyncInput.execute(ActorGraphInterpreter.scala:501)\n\
 at  akka.stream.impl.fusing.GraphInterpreterShell.processEvent(ActorGraphInterpreter.scala:599)\n\
 at  akka.stream.impl.fusing.ActorGraphInterpreter.akka$stream$impl$fusing$ActorGraphInterpreter$$processEvent(ActorGraphInterpreter.scala:768)\n\
 at  akka.stream.impl.fusing.ActorGraphInterpreter$$anonfun$receive$1.applyOrElse(ActorGraphInterpreter.scala:783)\n\
 at  akka.actor.Actor.aroundReceive(Actor.scala:534)\n\
 at  akka.actor.Actor.aroundReceive$(Actor.scala:532)\n\
 at  akka.stream.impl.fusing.ActorGraphInterpreter.aroundReceive(ActorGraphInterpreter.scala:690)\n\
 at  akka.actor.ActorCell.receiveMessage(ActorCell.scala:573)\n\
 at  akka.actor.ActorCell.invoke(ActorCell.scala:543)\n\
 at  akka.dispatch.Mailbox.processMailbox(Mailbox.scala:269)\n\
 at  akka.dispatch.Mailbox.run(Mailbox.scala:230)\n\
 at  akka.dispatch.Mailbox.exec(Mailbox.scala:242)\n\
 at  java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289)\n\
 at  java.util.concurrent.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1056)\n\
 at  java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1692)\n\
 at  java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:157)\n"
```

**Comment by jrudolph** [comment]
Fixed in #10321 and #10323.
